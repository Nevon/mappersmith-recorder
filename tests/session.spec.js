const MockDate = require("mockdate");
const { Session } = require("../");
const MockPersistence = require("./helpers/mockPersistence");

describe("Session", () => {
  beforeEach(() => {
    MockDate.set("2019-06-10T08:00:00.000Z");
  });

  afterEach(() => {
    MockDate.reset();
  });

  it("saves with the current timestamp as a suffix", async () => {
    const persistence = new MockPersistence();
    const session = new Session({ persistence, name: "test" });

    const filePath = await session.save();

    expect(persistence.getFiles()[filePath]).toMatchSnapshot();
  });
});
