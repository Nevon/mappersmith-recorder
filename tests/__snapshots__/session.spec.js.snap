// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session saves with the current timestamp as a suffix 1`] = `
Object {
  "log": Log {
    "browser": undefined,
    "comment": undefined,
    "creator": Creator {
      "comment": undefined,
      "name": "mappersmith-recorder",
      "version": "0.0.3",
    },
    "entries": Array [],
    "pages": Array [],
    "version": "1.2",
  },
}
`;
